src/test/regress/expected/coordinator_shouldhaveshards.out

```diff
@@ -778,6 +778,46 @@ SELECT table_name, citus_table_type, distribution_column, shard_count FROM publi
 (1 row)
 
 SET client_min_messages TO DEFAULT;
+-- issue 4508
+CREATE TABLE table_1 (key int, value text);
```

Comment:
```SQL
 WITH stats AS (
  SELECT count(key) m FROM table_1
),
inserts AS (
  INSERT INTO table_2
  SELECT key, count(*)
  FROM table_1
  WHERE key > (SELECT m FROM stats)
  GROUP BY key
  HAVING count(*) < (SELECT m FROM stats)
  LIMIT 1
  RETURNING *
) SELECT count(*) FROM inserts;
```

This query hits an assertion failure on `list_delete_cell_compat`. Can you check what is going on? The list seems fine, could it be a bug where we pass the parameters to `list_delete_cell_compat`?


(Also, lets add this to regression tests)